lib/collection/src/recommendations.rs

```diff
@@ -172,6 +173,16 @@ pub fn recommend_into_core_search(
     request: RecommendRequestInternal,
     all_vectors_records_map: &ReferencedVectors,
 ) -> CollectionResult<CoreSearchRequest> {
+    let query_request =
+        recommend_into_query_search(collection_name, request, all_vectors_records_map)?;
+    CoreSearchRequest::try_from(query_request)
+}
+
+pub fn recommend_into_query_search(
```

Comment:
This implementation means that for ungrouped recommend, we will transform from recommend -> query -> core. 

Wouldn't it be an easier mental model to not touch the recommend endpoint, and for groups, we can convert from recommend -> core -> query?